### ISSUE TITLE
clfft issues

### ISSUE THREAD

**Issue by tdd11235813** [mpicbg-scicomp/gearshifft#54]
- CPU only: 4096 with Inplace_Real and Outplace_Real yield `CL_OUT_OF_RESOURCES` when executing the backward plan (also reproduced it with a pure clfft code) (8192 or 2048 are working)
- it seems clfft is leaking memory
- I cannot handle clfft and out-of-memory kills, so I have to cap the input sizes to get the results in time

**Comment by psteinb** [comment]
thanks, let's keep this open for the record and concentrate on the GPUs 
on the clfft side for now. Feel free to report this to the clfft folks.

**Comment by tdd11235813** [comment]
the problem is, that the amount of leaked memory grows with input size and number of inputs. This explains why I could not handle the OOM kills after a while.
For some results for our draft I would separate the dimensions as I have done with fftw (1D, 2D and 3D have separate output csv files). If time allows, I prepare tomorrow a version for clfft folks.

**Comment by psteinb** [comment]
AFAIK, this was fixed with #55 ?

**Comment by tdd11235813** [comment]
yeah, I put an upper limit in #55, since clfft officially supports sizes to something 2^24.
The 4096 and the mem-leak is an open issue, see [issue 171](https://github.com/clMathLibraries/clFFT/issues/171) and [172](https://github.com/clMathLibraries/clFFT/issues/172)

**Comment by psteinb** [comment]
Ok, then let's wait until they are fixed. Thanks for the update.
